app/src/main/java/com/todoist_android/ui/home/BottomSheetEditTaskFragment.kt

```diff
@@ -0,0 +1,361 @@
+    private var newDueDate: String? = null
+    private var taskReminder: String? = null
+    private var taskStatus = "created"
+    private var userId: String? = null
+
+    companion object {
+        fun newInstance(item: TasksResponseItem): BottomSheetEditTaskFragment {
+            val bundle = Bundle()
+            bundle.apply {
+                putParcelable("data", item)
+            }
+            return BottomSheetEditTaskFragment().apply {
+                arguments = bundle
+            }
+        }
+    }
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        todoModel = arguments?.getParcelable("data")!!
+    }
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val dialog = super.onCreateDialog(savedInstanceState)
+        if (dialog is BottomSheetDialog) {
+            dialog.behavior.skipCollapsed = true
+            dialog.behavior.state = BottomSheetBehavior.STATE_EXPANDED
+        }
+        return dialog
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        binding = FragmentBottomsheetEditTaskBinding.inflate(layoutInflater, container, false)
+        return binding.root
+    }
+
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        setText()
+
+        setUpGlobalVariables()
+
+        getLoggedInUserId()
+
+        setOnClickListeners()
+
+    }
+
+    private fun setOnClickListeners(){
+        binding.ivEditReminder.setOnClickListener(this)
+        binding.ivEditFlag.setOnClickListener(this)
+        binding.tvEditDatePicker.setOnClickListener(this)
+        binding.tvDeleteTask.setOnClickListener(this)
+        binding.buttonEditTask.setOnClickListener(this)
+        binding.tvCloseEditTask.setOnClickListener(this)
+    }
+
+    private fun editTask(editTasksRequest: TodoModel) {
+        binding.root.hideKeyboard()
+        binding.pbEditBottomSheet.visibility = View.VISIBLE
+        Snackbar.make(dialog?.window!!.decorView, "Editing your task...", Snackbar.LENGTH_LONG)
+            .show()
+        viewLifecycleOwner.lifecycleScope.launchWhenStarted {
+            viewLifecycleOwner.repeatOnLifecycle(Lifecycle.State.STARTED) {
+                viewModel.editTasks(editTasksRequest).collect {
+                    when (it) {
+                        is APIResource.Success -> {
+                            binding.pbEditBottomSheet.visibility = GONE
+                            Snackbar.make(
+                                dialog?.window!!.decorView,
+                                "Task edited successfully",
+                                Snackbar.LENGTH_SHORT
+                            )
+                                .show()
+                            Log.d("task", editTasksRequest.toString())
+
```

Comment:
@Njumbi Trigger a refresh on the list, noticed after editing, on clicking the same item previous values are shown. list needs a refresh on item saved ... same for on Task deleted successfully

```
//trigger onRefresh in MainActivity
(activity as MainActivity).onRefresh()
```